% !TeX root = main.tex
\chapter{Material}
In this chapter we will have a look at the necessary computing infrastructure and the code base that I developed for studying the selection on loss-of-function variants. At the end of this chapter we will have a look at the \textit{A. thaliana} dataset we use.
\section{Computational Material and Resources}
In this section I will explain the computational material. We will start with the computing infrastructure that is necessary to perform our analysis and continue with looking at the location and dependencies of my code base.
\subsection{Computing Infrastructure}
The whole analysis is calculated on the institutes own high-performance computer cluster. With access to a computer node with 80 cores and 512 GB RAM. Also not all analysis need the maximum capacity of this computer node especially working with the basic unfiltered dataset consums a lot of RAM. To recapitulate the complete analysis access to a high-performance computer node is therefore necessary. 
\subsection{Code Base}
To study how premature stop codons get distributed in the genetic pool and shape the adaptation in plants we followed a completely bioinformatical based approach. Therefore we developed the statistical analysis and the data analysis in programming languages. To follow these ideas I use mainly python but also R as an additional language to complete the workflow. They are set up on an linux based system (Debian). If you like to recapitulate the analysis you can find the code in a github repository \url{https://github.com/laurasteinmann/Premature_Stop_Codons.git}.\\
Although at the moment there are no strict dependency on software versions I will list the used packages and there versions since there is the possibility of semantic changes in future versions. For the Python analysis part I use python (3.10) and the important packages for dealing with numerical data and big data analysis like pandas (1.4) and  numpy (1.22). For calculating statistical test I use scipy (1.8.0). For visualizing our results I use the matplotlib package (3.5), the seaborn package (0.11) and the matplotlib\_venn package (0.11). For dealing with genomic data and filtering it I use the R language with version 4.1.3 and the vcfR library (1.12)
\section{\textit{A. thaliana} dataset of 1001 Genomes Project}
The flowering plant \textit{Arabidopsis thaliana} is the standard reference plant for many disciplines in biology. For example for research questions in physiological, cellular, and molecular processes it is one of the most studied model organisms (Koornneef 2010\cite{Koornneef2010}). \textit{A. thaliana} especially gain importance for studying genes and determining their function since it was the first sequence of a plant genome in 2000 (Arabidopsis Genomes Initiative \cite{AGI2000}). With the relative small genome of 125 megabase pair that include about 25 000 genes \textit{A. thaliana} emphasized its importance as a model organism. This reference genome is based on the col-0 ecotype of \textit{A. thaliana} and its annotation and quality was improved with the technological progress since then.\\
\textit{A. thaliana} is also a promoted model organism for studying natural variation since \textit{A. thaliana} is not known for agricultural usage or as a crop plant. This means that in \textit{A. thaliana} we can still observe evolutionary selection and not the selection of human interests\cite{Koornneef2010}. By generating and publishing genomic sequences of 1, 135 ecotypes (accessions) of \textit{A. thaliana} in the 1001 Genomes project (1001 Genomes Consortium\cite{1001Genomes2016})  we have an ideal dataset to perform the analysis of adaptation in plant on population scale. These 1,135 accessions come from a worldwide hierarchical collection, which includes ecotypes from Sweden as well as the Iberian Peninsula as well as from North America and Central Asia. The accessions are a collection of naturally inbred lines that represent individuals under diverse ecological conditions. By sequencing with Illumina short read sequencing whole-genomes were systematically characterized of genome-wide polymorphisms. In the total collection 10,707,430 SNPs were identified. Based on this genomic information we can try to quantify genomic variation in a representative sample of accessions.\\
As an additional resource the 1001 Genomes Consortium provides 727 accessions with full-sequenced transcriptomes (Kawakatsu 2016\cite{Kawakatsu2016}). These accessions have an average expression of 18,000 genes and allows us to combine the information of genomic and transcriptomic data. Unfortunately these 727 accessions overlap not completely with the 1, 135 accessions. If we want to look at both datasets we can just base our analysis of 665 accessions. 
\chapter{Methods}
In this chapter we want to focus on the methods behind our analysis. We take a look at how the necessary statistical methods work and how we prepare the available information to extract the data we need to understand the selection in plant populations further. 
\section{Distribution of premature stop codons}
\section{Generation of a high confidential dataset}
\subsection{Classification of premature stop codons based on their gene expression differences}
\subsection{Classification of premature stop codons based on the calculation of the length of the remaining mRNA}