% !TeX root = main.tex
\chapter{Material}
In this chapter we will have a look at the necessary computing infrastructure and the code base that I developed for studying of the selection on loss-of-function variants. At the end of this chapter we will have a look at our used dataset.
\section{Computational Material and Resources}
In this section I will explain the computational material. We will start with the computing infrastructure we used for our analysis and continue with looking at the location and dependencies of our code base.
\subsection{Computing Infrastructure}
The whole analysis is calculated on the institutes own high-performance computer cluster. With access to a computer node with 80 cores and 512 GB RAM. Also not all analysis need the maximum capacity of this computer node especially working with the basic unfiltered dataset consums a lot of RAM. To recapitulate any analysis access to a high-performance computer node is therefore necessary. 
\subsection{Code Base}
To study how premature stop codons get distributed in the genetic pool and shape the adaptation in plants we followed a completely bioinformatical based approach. Therefore we developed the statistical analysis and the data analysis in programming languages. To follow these ideas use mainly python but also R as an additional language to complete the workflow set up on an linux based system (Debian). If you like to recapitulate the analysis you can find the code in a github repository \url{https://github.com/laurasteinmann/Premature_Stop_Codons.git}.\\
Although at the moment there are no strict dependency on software versions I will list the used packages and there versions since there is the possibility of semantic changes in future versions. For the Python analysis part we use python (3.10) and the important packages for dealing with numerical data and big data analysis like pandas (1.4) and  numpy (1.22). For calculating statistical test we use scipy (1.8.0). For visualizing our results we use the matplotlib package (3.5), the seaborn package (0.11) and the matplotlib\_venn package (0.11). For dealing with genomic data and filtering it we use the R language with version 4.1.3 and the vcfR library (1.12)
\section{Dataset of 1001 Genomes Project}
\chapter{Methods}